include/Mat.h

```diff
@@ -123,60 +121,66 @@ class Mat {
         return this->dims[ndims - 1];
     }
 
-    bool inbounds(size_type a){
-        if(a >= 0 && a < columns()) return true;
-        else return false;
-    }
-
-    bool inbounds(size_type a, size_type b){
-        if(a >= 0 && a < rows() && b >= 0 && b < columns()) return true;
-        else return false;
+    template<typename... arg>
+    bool inbounds(const arg... ind){
+        size_type temp[sizeof...(arg)] = {(static_cast<size_type>(ind))...};
+        for(int i = 0; i < ndims; i++){
+            if(temp[i] >= dims[i] || temp[i] < 0) return false;
+        }
+        return true;
     }
 
     bool isContiguous() const{
-        if(strides[0] != 1) return false;
-        if(ndims == 2 && strides[1] != dims[1]) return false;
-        else if(ndims > 2) errorCheck(true,
-            "n-dimensional contiguity check not yet implemented.");
+        size_type check = 1;
+        for(long i = ndims - 1; i >= 0; i--){
+            if(strides[i] != check) return false;
+            check *= dims[i];
+        }
         return true;
     }
 
-    Mat(size_type a = 1){
+    Mat(){
```

Comment:
1. Make default a shape = [0,] matrix.
2. Make sure we don't check memory==0 anymore